WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java

```diff
@@ -243,22 +271,23 @@ public void fetchCurrentTheme() {
                     @Override
                     public void onResponse(JSONObject response) {
                         try {
-                            mCurrentTheme = Theme.fromJSONV1_1(response, mSite);
-                            if (mCurrentTheme != null) {
-                                mCurrentTheme.setIsCurrent(true);
-                                mCurrentTheme.save();
-                                WordPress.wpDB.setCurrentTheme(String.valueOf(mSite.getSiteId()), mCurrentTheme.getId());
-                                if (mThemeBrowserFragment != null) {
-                                    mThemeBrowserFragment.setRefreshing(false);
-                                    if (mThemeBrowserFragment.getCurrentThemeTextView() != null) {
-                                        mThemeBrowserFragment.getCurrentThemeTextView().setText(mCurrentTheme.getName());
-                                        mThemeBrowserFragment.setCurrentThemeId(mCurrentTheme.getId());
-                                    }
-                                }
-                                if (mThemeSearchFragment != null && mThemeSearchFragment.isVisible()) {
-                                    mThemeSearchFragment.setRefreshing(false);
+                            mCurrentTheme = Theme.fromJSONV1_1(response, mSite.getSiteId());
+                            if (mCurrentTheme == null) {
+                                return;
+                            }
+                            mCurrentTheme.setIsCurrent(true);
+                            WordPress.wpDB.saveTheme(mCurrentTheme);
```

Comment:
Since this is a refactor PR, should we introduce a `ThemeTable` class and move the `saveTheme` & `setCurrentTheme` methods and all the theme related queries from `WordPressDB` there?